enf/domain.go

```diff
@@ -0,0 +1,79 @@
+package enf
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+// Domain represents a domain in the ENF.
+type Domain struct {
+	Name    *string `json:"name"`
+	Network *string `json:"network"`
+	Status  *string `json:"status"`
+}
+
+// DomainRequest represents a request to provision a new domain.
+type DomainRequest struct {
+	Name       *string `json:"name"`
+	Type       *string `json:"type"`
+	AdminName  *string `json:"admin_name"`
+	AdminEmail *string `json:"admin_email"`
+}
+
+type domainResponse struct {
+	Data []*Domain              `json:"data"`
+	Page map[string]interface{} `json:"page"`
+}
+
+// ListDomains lists all available domains on the ENF.
+func (s *DomainService) ListDomains(ctx context.Context) ([]*Domain, *http.Response, error) {
+	path := fmt.Sprintf("api/xcr/v2/domains")
+	body, resp, err := s.client.get(ctx, path, new(domainResponse))
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return body.(*domainResponse).Data, resp, nil
+}
+
+// GetDomain gets the information of a specified domain.
+func (s *DomainService) GetDomain(ctx context.Context, domain string) (*Domain, *http.Response, error) {
+	path := fmt.Sprintf("api/xcr/v2/domains/%v", domain)
+	body, resp, err := s.client.get(ctx, path, new(domainResponse))
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return body.(*domainResponse).Data[0], resp, nil
+}
+
+// CreateDomain provisions a domain on the ENF.
+func (s *DomainService) CreateDomain(ctx context.Context, req *DomainRequest) (*Domain, *http.Response, error) {
+	path := fmt.Sprintf("api/xcr/v2/domains")
+	body, resp, err := s.client.post(ctx, path, new(domainResponse), req)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return body.(*domainResponse).Data[0], resp, nil
+}
+
+// ActivateDomain activates the given domain (sets the status field to ACTIVE)
+func (s *DomainService) ActivateDomain(ctx context.Context, domain string) (*Domain, *http.Response, error) {
+	path := fmt.Sprintf("api/xcr/v2/domains/%v/status", domain)
+	// The ActivateDomain endpoint will accept any request body, so there's no need to have a *DomainRequest parameter.
+	// Instead, just use an empty DomainRequest struct as the "body".
+	body, resp, err := s.client.put(ctx, path, new(domainResponse), DomainRequest{})
```

Comment:
`DomainRequest` isn't related to this API call, so using an empty one as the "body" feels weird.  How about an [empty struct](https://medium.com/@l.peppoloni/how-to-improve-your-go-code-with-empty-structs-3bd0c66bc531) instead?

```
body, resp, err := s.client.put(ctx, path, new(domainResponse), struct{}{})
```

Reply:
I didn't even consider this, but you're totally right. An empty struct is more natural.